kratos/processes/entity_erase_process.cpp

```diff
@@ -0,0 +1,162 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/kratos_flags.h"
+#include "processes/entity_erase_process.h"
+#include "utilities/variable_utils.h"
+#include "utilities/auxiliar_model_part_utilities.h"
+
+namespace Kratos
+{
+template<class TEntity>
+const Kratos::Flags EntitiesEraseProcess<TEntity>::REMOVE_FROM_ALL_LEVELS(Kratos::Flags::Create(0));
+template<class TEntity>
+const Kratos::Flags EntitiesEraseProcess<TEntity>::NOT_REMOVE_FROM_ALL_LEVELS(Kratos::Flags::Create(0, false));
+template<class TEntity>
+const Kratos::Flags EntitiesEraseProcess<TEntity>::ASSIGN_FLAG(Kratos::Flags::Create(1));
+template<class TEntity>
+const Kratos::Flags EntitiesEraseProcess<TEntity>::NOT_ASSIGN_FLAG(Kratos::Flags::Create(1, false));
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+EntitiesEraseProcess<TEntity>::EntitiesEraseProcess(
+    ModelPart& rModelPart,
+    Flags Options
+    )
+    : mrModelPart(rModelPart),
+     mrOptions(Options)
+{
+    KRATOS_TRY
+
+    KRATOS_CATCH("")
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<class TEntity>
+EntitiesEraseProcess<TEntity>::EntitiesEraseProcess(
+    ModelPart& rModelPart,
+    Parameters ThisParameters
+    )
+    : mrModelPart(rModelPart)
+{
+    KRATOS_TRY
+
+    // The default parameters
+    Parameters default_parameters = GetDefaultParameters();
+    ThisParameters.RecursivelyValidateAndAssignDefaults(default_parameters);
+
+    // Assign parameters
+    mrOptions.Set(REMOVE_FROM_ALL_LEVELS, ThisParameters["remove_from_all_levels"].GetBool());
+    mrOptions.Set(ASSIGN_FLAG, ThisParameters["assign_flag"].GetBool());
```

Comment:
Shouldn't be `REMOVE_ALL_ENTTITES`?

Reply:
Or `ERASE_ALL_ENTITIES`?

Reply:
OK, any preference

Reply:
Just mentioning that the process is EntityEraseProcess but the flags are some with erase and some with remove. I would uinfy them.

Reply:
It is done